File: HackerRank Solutions/10 Days of Statistics/Day 1 Quartiles.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
from statistics import median

n = int(input())
Q2 = [int(x) for x in input().split()]
Q2.sort()
t = int(len(Q2)/2)
if(len(Q2)%2 == 0):
    Q1 = Q2[:t]
    Q3 = Q2[t:]
else:
    Q1 = Q2[:t]
    Q3 = Q2[t+1:]
print(int(median(Q1)))
print(int(median(Q2)))
print(int(median(Q3)))

File: HackerRank Solutions/10 Days of Statistics/Day 1 Quartiles/Day 1 Quartiles.java
import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();
        int [] x = new int[n];
        for(int i=0; i<n; i++){
            x[i] = scan.nextInt();
        }
        Arrays.sort(x);
        System.out.println(getMedian(x, 0, n/2 - 1));
        System.out.println(getMedian(x, 0, n-1));
        if(n%2 == 0){
            System.out.println(getMedian(x, n/2, n-1));
        } else{
            System.out.println(getMedian(x, n/2 + 1, n-1));
        }
    }
    private static int getMedian(int[] a, int start, int end){
        int median = 0;
        int size = end - start + 1;
        if(size%2 == 0){
            median = (a[start + size/2 - 1] + a[start + size/2]) / 2;
        } else{
            median = a[start + (size -1)/2];
        }
        return median;
    }
}